# Analisis de prestaciones por diagnostico

- En este apartado sequiere ver la cantidad de prestaciones que se realizan por cada diagnostico principal.

- El resultado que se quiere obtener es parecido al siguiente:

| Año  | Diagnostico | Resumen Procedimientos     |
| ---- | ----------- | -------------------------- |
| 2019 | "I25.1"     | {"45.23": 20, "73.59": 10} |
| 2020 | "I25.1"     | {"45.23": 7, "73.59": 3}   |


In [1]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2


In [2]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import warnings

from src.features import build_features

warnings.filterwarnings("ignore")
sns.set_style()
plt.rcParams["figure.figsize"] = (12, 6)

HOSPITAL_DEL_TORAX = 112103

## 1. Análisis de Procedimientos a Nivel País

En este análisis se quieren responder las siguiente preguntas:

1. ¿Cómo es el ranking de los hospitales para cada uno de los procedimientos realizads?
2. ¿Quién es el Hospital que realiza más de X procedimientos?
3. ¿El procedimiento X es realizado en todo el país? ¿En qué partes en específico se realiza?

Para responder esta pregunta se utilizará la base de datos de Egresos de Hospitales adscritos
a GRD entre 2019 y 2021.


In [3]:
df_procesada = pd.read_csv("../data/processed/df_procesada.csv", sep=";")
df_procesada["ANIO_EGRESO"] = df_procesada["ANIO_EGRESO"].astype("Int32")

In [5]:
display(df_procesada.head(10))

COD_HOSPITAL  CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO  ETNIA   PROVINCIA  \
0        118100       1314867.0   MUJER       1988-06-19  OTRO   CONCEPCION   
1        118100       1354418.0   MUJER       1988-03-27  OTRO   CONCEPCION   
2        118100        239861.0   MUJER       2017-06-17  OTRO   CONCEPCION   
3        118100        330326.0  HOMBRE       2017-12-04  OTRO   CONCEPCION   
4        118100       1369293.0   MUJER       1995-12-06  OTRO   CONCEPCION   
5        118100        469385.0   MUJER       1942-06-29  OTRO   CONCEPCION   
6        118100       1146635.0   MUJER       1951-07-14  OTRO   CONCEPCION   
7        118100         93557.0  HOMBRE       1938-03-04  OTRO   CONCEPCION   
8        118100        674943.0   MUJER       1962-08-21  OTRO   CONCEPCION   
9        118100       1079955.0  HOMBRE       1944-12-07  OTRO   CONCEPCION   

                COMUNA NACIONALIDAD                       PREVISION  \
0  SAN PEDRO DE LA PAZ        CHILE  FONASA INSTITUCIONAL - (MAI) A   
1  SAN PEDRO DE LA PAZ        CHILE  FONASA INSTITUCIONAL - (MAI) D   
2           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   
3           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) A   
4              FLORIDA        CHILE  FONASA INSTITUCIONAL - (MAI) C   
5          CHIGUAYANTE        CHILE  FONASA INSTITUCIONAL - (MAI) B   
6           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   
7                 LOTA        CHILE  FONASA INSTITUCIONAL - (MAI) D   
8          SANTA JUANA        CHILE  FONASA INSTITUCIONAL - (MAI) B   
9           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   

  SERVICIO_SALUD  ... USOSPABELLON IR_29301_COD_GRD IR_29301_PESO  \
0     CONCEPCIÓN  ...          1.0         146102.0        0.5744   
1     CONCEPCIÓN  ...          NaN         134161.0        0.2951   
2     CONCEPCIÓN  ...          NaN          64182.0        0.6736   
3     CONCEPCIÓN  ...          NaN          64172.0        0.5475   
4     CONCEPCIÓN  ...          NaN         134141.0        0.3107   
5     CONCEPCIÓN  ...          NaN         114103.0        1.1799   
6     CONCEPCIÓN  ...          NaN          44163.0        1.0328   
7     CONCEPCIÓN  ...          1.0          11101.0        1.8731   
8     CONCEPCIÓN  ...          1.0          61402.0        1.3078   
9     CONCEPCIÓN  ...          NaN          14133.0        1.6687   

  IR_29301_SEVERIDAD IR_29301_MORTALIDAD  \
0                2.0                 1.0   
1                1.0                 1.0   
2                2.0                 2.0   
3                2.0                 2.0   
4                1.0                 1.0   
5                3.0                 3.0   
6                3.0                 3.0   
7                1.0                 1.0   
8                2.0                 2.0   
9                3.0                 3.0   

                          HOSPPROCEDENCIA ESTANCIA ANIO_EGRESO MES_EGRESO  \
0                                     NaN      5.0      2019.0        3.0   
1                                     NaN      4.0      2019.0        3.0   
2                                     NaN      6.0      2019.0        3.0   
3                                     NaN      2.0      2019.0        3.0   
4                                     NaN      6.0      2019.0        3.0   
5                                     NaN     47.0      2019.0        3.0   
6                                     NaN     43.0      2019.0        3.0   
7                                     NaN     41.0      2019.0        3.0   
8  HOSPITAL CLORINDA AVELLO (SANTA JUANA)     33.0      2019.0        3.0   
9                                     NaN     31.0      2019.0        3.0   

    FECHA  
0  2019-3  
1  2019-3  
2  2019-3  
3  2019-3  
4  2019-3  
5  2019-3  
6  2019-3  
7  2019-3  
8  2019-3  
9  2019-3  

[10 rows x 133 columns]

Los procedimientos en esta base de datos se encuentran en las columnas "PROCEDIMIENTOSX", donde X
va desde 1 hasta 30. Para analizar esta base de forma más expedita, se transformará de formato
ancho a formato largo, utilizando como identificador a las columnas ["COD_HOSPITAL", "ANIO_EGRESO",
"CIP_ENCRIPTADO", "DIAGNOSTICO1"]. Cabe destacar que la columna "CIP_ENCRIPTADO" hace referencia al
RUT del paciente.


In [13]:
procedimientos_formato_long = pd.melt(
    df_procesada,
    id_vars=["COD_HOSPITAL", "ANIO_EGRESO", "CIP_ENCRIPTADO", "DIAGNOSTICO1"],
    value_vars=[f"PROCEDIMIENTO{i}" for i in range(1, 31)],
    value_name="procedimiento",
)

In [27]:
conteo_y_ranking_procedimientos = build_features.contar_procedimientos_y_rankear(
    procedimientos_formato_long
)

In [28]:
display(conteo_y_ranking_procedimientos.head(20))

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo  ranking  \
150206       2021.0        108100         99.99       2        1   
165765       2021.0        113180         99.99       2        2   
140654       2021.0        103100         99.99       1        3   
143682       2021.0        105100         99.99       1        4   
155556       2021.0        110120         99.99       1        5   
167921       2021.0        114101         99.99       1        6   
183553       2021.0        121109         99.99       1        7   
186513       2021.0        122100         99.99       1        8   
190822       2021.0        126100         99.99       1        9   
191999       2021.0        129106         99.98      19        1   
151485       2021.0        109100         99.98      11        2   
186788       2021.0        123100         99.98      11        3   
165565       2021.0        113180         99.98       8        4   
178447       2021.0        118100         99.98       8        5   
181258       2021.0        120101         99.98       8        6   
153822       2021.0        110100         99.98       7        7   
176484       2021.0        117101         99.98       6        8   
182651       2021.0        121109         99.98       6        9   
154968       2021.0        110120         99.98       5       10   
156780       2021.0        111100         99.98       5       11   

        diferencia_proximo_hospital  total_procedimientos  
150206                            0                    11  
165765                            1                    11  
140654                            0                    11  
143682                            0                    11  
155556                            0                    11  
167921                            0                    11  
183553                            0                    11  
186513                            0                    11  
190822                          -18                    11  
191999                            8                   119  
151485                            0                   119  
186788                            3                   119  
165565                            0                   119  
178447                            0                   119  
181258                            1                   119  
153822                            1                   119  
176484                            0                   119  
182651                            1                   119  
154968                            0                   119  
156780                            0                   119

El resultado muestra el ranking para todos los procedimientos asignados en el país en cada uno de 
los años analizados (2019 a 2021). 

A modo de ejemplo, se ilustra el ranking en el año 2021 para el procedimiento 99.99. En este caso,
se observa que los hospitales con código 108100 y 113180 están en los primeros lugares, realizando
2 procedimientos cada uno.

In [23]:
mejores_procedimientos_torax = conteo_y_ranking_procedimientos.query(
    "COD_HOSPITAL == @HOSPITAL_DEL_TORAX and ranking == 1"
)

In [26]:
conteo_y_ranking_procedimientos.query("procedimiento == 35.05")

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo  ranking  \
178208       2021.0        118100         35.05      19        1   
153623       2021.0        110100         35.05      18        2   
162336       2021.0        112103         35.05      13        3   
179991       2021.0        119100         35.05       9        4   
147813       2021.0        107100         35.05       6        5   
166735       2021.0        114101         35.05       6        6   
139732       2021.0        103100         35.05       5        7   
158914       2021.0        111195         35.05       5        8   
151659       2021.0        109100         35.05       4        9   
190325       2021.0        126100         35.05       2       10   
181778       2021.0        120101         35.05       1       11   
88627        2020.0        110100         35.05      13        1   
119103       2020.0        118100         35.05      12        2   
81750        2020.0        107100         35.05       6        3   
92592        2020.0        111100         35.05       6        4   
104945       2020.0        114101         35.05       5        5   
71888        2020.0        103100         35.05       2        6   
77560        2020.0        105101         35.05       1        7   
99115        2020.0        112102         35.05       1        8   

        diferencia_proximo_hospital  total_procedimientos  
178208                          1.0                    88  
153623                          5.0                    88  
162336                          4.0                    88  
179991                          3.0                    88  
147813                          0.0                    88  
166735                          1.0                    88  
139732                          0.0                    88  
158914                          1.0                    88  
151659                          2.0                    88  
190325                          1.0                    88  
181778                         -2.0                    88  
88627                           1.0                    46  
119103                          6.0                    46  
81750                           0.0                    46  
92592                           1.0                    46  
104945                          3.0                    46  
71888                           1.0                    46  
77560                           0.0                    46  
99115                          -2.0                    46

In [7]:
mejores_procedimientos_torax.query("diferencia_proximo_hospital == -1")

ANIO_EGRESO  COD_HOSPITAL  value  conteo  ranking  \
162597       2021.0        112103  87.93       1        1   
162525       2021.0        112103  32.21       1        1   
99861        2020.0        112103  40.69       1        1   
99661        2020.0        112103   37.6       5        1   
99837        2020.0        112103  37.37       1        1   
99732        2020.0        112103  33.48       2        1   

        diferencia_proximo_hospital  
162597                         -1.0  
162525                         -1.0  
99861                          -1.0  
99661                          -1.0  
99837                          -1.0  
99732                          -1.0